com.ibm.streamsx.utility/com.ibm.streamsx.utility/native.function/utility.cpp

```diff
@@ -104,4 +105,10 @@ namespace com { namespace ibm { namespace streamsx { namespace utility {
         if (rc<0) THROW (SPLRuntimeOperator, "could not set processor affinity to CPUs"
             << ", " << strerror(errno));
     }
+
+    void setThreadName(SPL::rstring name)
+    {
+        pthread_setname_np(pthread_self(), name.c_str());
```

Comment:
Can we add error checking? `pthread_setname_np()` can return `ERANGE` to indicate that the name is too long (more than 16 characters, including `NULL` byte).

I think it's also worth pointing out that `pthread_setname_np()` is not supported on all of our platforms, which I found out the hard way. I think it relies on a version of glibc that RHEL6 doesn't have. I'm actually okay with that, as I assume platforms will pick up later versions of glibc. I got around this by using `prctl()` with the flag `PR_SET_NAME`.